### PR TITLE
Add thread to check drain status + AgentStatusWatcher unittest

### PR DESCRIPTION
**Pull request by goughes**
Adding the skeleton thread for checking the drain status.  Need to figure out what database connections are needed and what drain info to write where.

### CODE REVIEW

**Comment by amaltaro** [comment]
@goughes can you rebase it please? I'll review and run some tests tomorrow

**Comment by goughes** [comment]
Well I don't think I did that right... I'll fix it.

**Comment by goughes** [comment]
@amaltaro Take a look and let me know if that's what you wanted.

**Comment by ticoann** [comment]
Hi Erik, there is still conflict on the current master. You would need to merge with master and rebase.

**Comment by ticoann** [comment]
test this please

**Review by amaltaro — CHANGES_REQUESTED** [review]
Since we'll have to merge this drainInfo with all the information collected by AgentStatusWatcher, we'll have to somehow expose it to other classes.

Thinking on the best - and most pythonic - way of accomplishing it, how about we define a class variable and we access it via a classmethod (getter)? @ticoann @goughes any better idea?

**Review by amaltaro — COMMENTED** [review]
Changes look good now, Erik. Can you please fix those unit tests that are failing (I couldn't see a relationship with these changes, but...)?
Once that's fixed, we can test it in the agent itself.

**Comment by goughes** [comment]
test this please

**Comment by goughes** [comment]
The unittests that Jenkins says are failing work on my local machine using Docker.  Can someone give this a final review?  Let me know how important fixing the Pylint and Python3 tests are.

**Comment by amaltaro** [comment]
test this please

**Comment by amaltaro** [comment]
test this please

**Comment by amaltaro** [comment]
@goughes the pylint unused imports warnings have to be fixed. Also fix the string format message please (instead of % it's comma). Others are not really fixable or take too much effort (like methods signature and arguments not used there).

The python3 import suggestions are supposed to be added to any new files.

However, the most important is understand and fixing the unit tests. BTW, please make a separate commit for all these pylint/pep8/unittests/code reformat stuff, such that we don't bother reviewing that, but only the actual code.

**Comment by amaltaro** [comment]
test this please

**Comment by amaltaro** [comment]
test this please

File: src/python/WMComponent/AgentStatusWatcher/DrainStatusPoller.py
Comment (amaltaro) [inline src/python/WMComponent/AgentStatusWatcher/DrainStatusPoller.py]: this traceback line can be removed (and the import). the logging.exception line below will automatically print the traceback.

**Comment by amaltaro** [comment]
test this please

**Comment by amaltaro** [comment]
@goughes did you manage to run a couple of workflows with this code? If so, then I think we're ready to merge it (after a minor comment I made yesterday)

**Comment by goughes** [comment]
test this please

**Comment by amaltaro** [comment]
test this please

**Comment by amaltaro** [comment]
test this please

**Comment by amaltaro** [comment]
I haven't seen any issues with this patch. I think it's good to be merged.
We still have to backport it to 1.1.4 wmagent branch, but I guess that can wait until we're about to cut the production release.

**Comment by ticoann** [comment]
@goughes, Erik, could you squash the commit? I will merge the patch. Thanks,

**Comment by amaltaro** [comment]
[Fix #7615 - Add thread to check drain status + AgentStatusWatcher unittest](https://trello.com/c/FTcLfCmV/40-fix-7615-add-thread-to-check-drain-status-agentstatuswatcher-unittest)

**Comment by ticoann** [comment]
test this please

**Comment by ticoann** [comment]
test this please

**Comment by amaltaro** [comment]
How come it constantly fails... I cannot find anything, but breaking all these unit tests is not going to be good

**Comment by amaltaro** [comment]
test this please

**Comment by amaltaro** [comment]
@goughes or @ticoann would you mind running all the unittests for following files (maybe run 3 times each):
```
test/python/WMComponent_t/JobAccountant_t $ nosetests JobAccountant_t.py 
test/python/WMComponent_t/RetryManager_t $ nosetests RetryManager_t.py 
```

JobAccountant fails 1 out of 15 tests consistently. RetryManager is 100% success. I ran it on my VM, but that one is using oracle backend, which might be yielding different results than mysql/jenkins. Though I think `'NoneType' object has no attribute 'engine'` would affect both.

**Comment by amaltaro** [comment]
@goughes @ticoann I think I got lost with this PR.
Since the drain status thread has been merged already in another PR https://github.com/dmwm/WMCore/pull/7955

is there anything else that we want from this PR? Maybe Erik should close it and start a fresh new one?

**Comment by goughes** [comment]
@amaltaro @ticoann Closing this one.  I'll start a fresh PR for an AgentStatusWatcher unit test that doesn't use the emulators.

**Pull request closed without merging**
